### DISCUSSION TITLE
Support: Add `Cancel` and `Save` button both in the Form

### DISCUSSION THREAD

**Discussion by ahmedsakri** [manojadams/mui-forms#47] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I need to show cancel and save button in form always. When clicked on cancel it should close the form and click on save/submit submit the filled form. Anyway i can achieve that!

**Describe the solution you'd like**
Can we pass cancel button in buttons array in JSON schema and it will have these functionalities?

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by manojadams** [comment]
@ahmedsakri Is this requirement of a simple form or grouped form (like tabs)?

**Comment by manojadams** [comment]
@ahmedsakri Here is how you can pass your custom buttons for next, previous and submit action:

```
      <MuiForms 
          ...
          ...
          buttons={{
              next: <Button variant="contained">Next</Button>,
              previous: <Button variant="outlined">Previous</Button>,
              submit: <Button variant="contained">Login</Button>
          }}
      />
```

**Comment by manojadams** [comment, marked as answer]
@ahmedsakri This feature is now available wherein you can provide your custom footer component to achieve the same:
https://mui-forms.vercel.app/custom-footer

**Reply by venkatesh1146** [reply]
Hi @manojadams 

What version of mui forms do i need to install go get this custom footer feature?

Currently i am using `"mui-forms": "2.0.21",`
